Ingestor + projector cxi

JulReinhardt opened the pull request.

JulReinhardt committed:
following examplePlugin

JulReinhardt committed:
changes discussed with Ian

JulReinhardt committed:
add filter operations and import from shared_operations

JulReinhardt committed:
ignore operations

JulReinhardt committed:
begin cxi ingestor

JulReinhardt committed:
comment in ingestor

JulReinhardt committed:
ignore

JulReinhardt committed:
comments

JulReinhardt committed:
Merge branch 'master' of https://github.com/Xi-CAM/Xi-cam.spectral into ingestor_pystxm

JulReinhardt committed:
Merge branch 'operations_pystxm' into ingestor_pystxm

# Conflicts:
#	.gitignore
#	xicam/spectral/__init__.py

JulReinhardt committed:
Merge branch 'master' of https://github.com/Xi-CAM/Xi-cam.spectral

JulReinhardt committed:
work on ingestor for new cxi files from David

JulReinhardt committed:
Merge branch 'master' of https://github.com/Xi-CAM/Xi-cam.spectral

JulReinhardt committed:
move NXcxi ingestor to seperate file

JulReinhardt committed:
Merge branch 'master' into ingestor_pystxm

# Conflicts:
#	.gitignore
#	xicam/spectral/__init__.py
#	xicam/spectral/ingestors/__init__.py

JulReinhardt committed:
calc coords from pixelsize

JulReinhardt committed:
imports

JulReinhardt committed:
re-ordering code

JulReinhardt committed:
calc pixel size (TEMP solution) and sort xarray

JulReinhardt committed:
clean up

JulReinhardt committed:
clean up

JulReinhardt commented:
If either of you could have a look at the cxi ingestor and make comments, I'd appreciate a discussion about:

- [ ] how to handle multiple data sets to be compared (as one --> rescaling pixelsizes, sorting energies etc. probably not done in the ingestor)
- [ ] naming conventions for projections
- [ ] naming of the frame_data_keys since 'raw' is not actually true for reconstructed ptychographic images

JulReinhardt committed:
minor fixes

dylanmcreynolds commented on .gitignore:
```diff
@@ -7,4 +7,6 @@
\ No newline at end of file
+*.egg-info/
+.vscode
+venv
```
Doesn't this need to be 'venv/'?

JulReinhardt replied:
probably yes, don't how that sneaked in there

dylanmcreynolds commented on xicam/spectral/ingestors/NXcxi_ptycho.py:
```diff
@@ -0,0 +1,110 @@
+    dim_x = rec_shape[0]
+    dim_y = rec_shape[1]
+
+    def rec_psize_nm(energy, corner_x, corner_y, corner_z):
```
Is there a reason this is a nested function?

JulReinhardt replied:
Probably not, only my sloppy writing ...

dylanmcreynolds reviewed (commented):
I have a sample pytest for an ingestor that you might consider looking at for the pystxm ingestor. It's not released, but can be found: https://github.com/als-computing/beamline-data/blob/mapped_ingestor/docstream/tests/test_docstream.py

JulReinhardt committed:
add ingestor entrypoint

JulReinhardt commented:
> These entry points look odd to me. If we're registering them xi-cam-wide, how do we map from one person's application/x-hdf5 or x-fits to another's idea of what those content types mean? I assume some code somewhere is mapping file extension to content type? Or am I completely misreading that?

I don't know, that was one of my questions that still needs to be addressed. @ronpandolfi can you comment, when you have a moment for this?

JulReinhardt committed:
new  projections

JulReinhardt committed:
dask array for recs only

JulReinhardt committed:
import ingest function

JulReinhardt committed:
minor

JulReinhardt committed:
comment out library modules to avoid conflicts

JulReinhardt committed:
clean up ingestor cxi ptycho

JulReinhardt committed:
add projector

JulReinhardt committed:
add fields and streams to projection

JulReinhardt committed:
only use abs part of complex reconstruction for now

JulReinhardt committed:
read correct coords fields and use NXcxi projector for now

JulReinhardt committed:
handling multiple projectors

JulReinhardt committed:
intermediate working state to show phase and transmission stream in catalogviewer and  transmission in spectral

JulReinhardt committed:
add (preliminary version for) ImageHint

JulReinhardt committed:
small fxes

JulReinhardt committed:
minors and comments

ronpandolfi committed:
Merge branches 'ingestor_pystxm' and 'master' of https://github.com/Xi-cam/Xi-cam.spectral into ingestor_pystxm

# Conflicts:
#	setup.py
#	xicam/spectral/__init__.py
#	xicam/spectral/ingestors/__init__.py

ronpandolfi committed:
Merge branches 'ingestor_pystxm' and 'master' of https://github.com/Xi-cam/Xi-cam.spectral into ingestor_pystxm

# Conflicts:
#	setup.py
#	xicam/spectral/__init__.py
#	xicam/spectral/ingestors/__init__.py

ronpandolfi committed:
Make dim strings consistent
Merge pull request #4 from Xi-CAM/tv_mode

Refactor TV mode to use callbacks rather than hot-polling
Merge remote-tracking branch 'origin/vis-refactor' into vis-refactor

# Conflicts:
#	setup.py
Merge branch 'master' of https://github.com/Xi-CAM/Xi-cam into nd-imageviewer

# Conflicts:
#	xicam/gui/tests/test_ndimageview.py

ronpandolfi commented:
> These entry points look odd to me. If we're registering them xi-cam-wide, how do we map from one person's application/x-hdf5 or x-fits to another's idea of what those content types mean? I assume some code somewhere is mapping file extension to content type? Or am I completely misreading that?

Its an unresolved issue that there's no good way to dispatch to different ingestors associated with the same mimetype. We've had some ideas, but none that I'm fully satisfied with yet.

ronpandolfi committed:
Fix gitignore path

ronpandolfi committed:
use nexus app def name as key

ronpandolfi committed:
Remove .vscode
